x/mongo/driver/crypt.go

```diff
@@ -382,3 +389,69 @@ func (c *crypt) decryptKey(kmsCtx *mongocrypt.KmsContext) error {
 		}
 	}
 }
+
+// needsKmsProvider returns true if provider was initially set to an empty document.
+// An empty document signals the driver to fetch credentials.
+func needsKmsProvider(kmsProviders *bsoncore.Document, provider string) bool {
+	gcpVal, err := kmsProviders.LookupErr(provider)
```

Comment:
Is this always a GCP provider? If not, the variable name `gcpVal` is somewhat confusing.